content/docs/2.0/scalers/azure-log-analytics.md

```diff
@@ -0,0 +1,150 @@
++++
+title = "Azure Log Analytics"
+layout = "scaler"
+availability = "v2.0+"
+maintainer = "Community"
+description = "Scale applications based on Azure Log Analytics query result"
+go_file = "azure_log_analytics_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `azure-log-analytics` trigger for Azure Log Analytics query result (All auth parameters in metadata is not real, generated randomly).
+
+```yaml
+triggers:
+  - type: azure-log-analytics
+    metadata:
+      tenantID: "72f988bf-86f1-41af-91ab-2d7cd011db47"
+      clientID: "04b4ca0a-82b1-4c0a-bbbb-7946442e805b"
+      clientSecret: "vU6UtUXls6RNXxv~l6NRi1V8J1fnk5Q-ce"
+      workspaceID: "81963c40-af2e-47cd-8e72-3002e08aa2af"
+      query: "let AppName = \"web\";\r\nlet ClusterName = \"demo-cluster\";\r\nlet AvgDuration = ago(10m);\r\nlet ThresholdCoefficient = 0.8;\r\nPerf\r\n| where InstanceName contains AppName\r\n| where InstanceName contains ClusterName\r\n| where CounterName == \"cpuUsageNanoCores\"\r\n| where TimeGenerated > AvgDuration\r\n| extend AppName = substring(InstanceName, indexof((InstanceName), \"/\", 0, -1, 10) + 1)\r\n| summarize MetricValue=round(avg(CounterValue)) by CounterName, AppName\r\n| join (Perf \r\n        | where InstanceName contains AppName\r\n        | where InstanceName contains ClusterName\r\n        | where CounterName == \"cpuLimitNanoCores\"\r\n        | where TimeGenerated > AvgDuration\r\n        | extend AppName = substring(InstanceName, indexof((InstanceName), \"/\", 0, -1, 10) + 1)\r\n        | summarize arg_max(TimeGenerated, *) by AppName, CounterName\r\n        | project Limit = CounterValue, TimeGenerated, CounterPath, AppName)\r\n        on AppName\r\n| project MetricValue, Threshold = Limit * ThresholdCoefficient"

```

Comment:
I'm wondering - Do we use in-line queries, use config map or mount them on the scaled workload?

Reply:
Thoughts @zroubalik @ahmelsayed ?

Reply:
@tomkerkhove , I used Prometheus Scaler as reference: https://keda.sh/docs/1.5/scalers/prometheus/

Reply:
Makes sense but in this case, they are a bit longer. It's ok for me but just want to verify with the others.

Reply:
You can define a multi-line string in yaml using 

```yaml
      ...
      query: |
        let AppName = "web";
        let ClusterName = "demo-cluster";
        let AvgDuration = ago(10m);
        let ThresholdCoefficient = 0.8;
        Perf
        | where InstanceName contains AppName
        | where InstanceName contains ClusterName
        | where CounterName == "cpuUsageNanoCores"
        | where TimeGenerated > AvgDuration
        | extend AppName = substring(InstanceName, indexof((InstanceName), "/", 0, -1, 10) + 1)
        | summarize MetricValue=round(avg(CounterValue)) by CounterName, AppName
        | join (Perf 
                | where InstanceName contains AppName
                | where InstanceName contains ClusterName
                | where CounterName == "cpuLimitNanoCores"
                | where TimeGenerated > AvgDuration
                | extend AppName = substring(InstanceName, indexof((InstanceName), "/", 0, -1, 10) + 1)
                | summarize arg_max(TimeGenerated, *) by AppName, CounterName
                | project Limit = CounterValue, TimeGenerated, CounterPath, AppName)
                on AppName
        | project MetricValue, Threshold = Limit * ThresholdCoefficient
```

which might make it look a bit more readable.

Reply:
Love the suggestion!